### DISCUSSION TITLE
Question about passing both w3c style and jsonwp style caps to the server

### DISCUSSION THREAD

**Discussion by lukyth** [webdriverio/webdriverio#6563] [Technical Discussion]
https://github.com/webdriverio/webdriverio/blob/e9b565e0cb6aeca69d331cbd883734ec72af04f1/packages/webdriver/src/utils.ts#L50-L73

> the user could have passed in either w3c style or jsonwp style caps and we want to pass both styles to the server

I wonder what's the reason that both styles should be passed, not just only one based on what the user passed?

So I'm thinking about something like this:
```js
const body = params.capabilities && (params.capabilities as Capabilities.W3CCapabilities).alwaysMatch
    ? { capabilities: params.capabilities }
    : { desiredCapabilities: params.capabilities }
const sessionRequest = new WebDriverRequest('POST', '/session', body)
```

**Comment by christian-bromann** [comment]
> I wonder what's the reason that both styles should be passed, not just only one based on what the user passed?

`alwaysMatch/firstMatch` is a protocol detail we have hidden from the user as it is not wide known and usually not commonly used. In order to have a frictionless upgrade to W3C conform remote endpoints we have send both types of caps. This has worked quite well so far.

**Reply by lukyth** [reply]
Totally make sense that it'll be easier for the user to use this without specifying `alwaysMatch/firstMatch`. I understand the reason behind this now. Thank you so much! 👍
